src/index/ivf/ivf.cc

```diff
@@ -116,8 +136,9 @@ class IvfIndexNode : public IndexNode {
     Deserialize(const BinarySet& binset, const Config& config) override;
     Status
     DeserializeFromFile(const std::string& filename, const Config& config) override;
-    std::unique_ptr<BaseConfig>
-    CreateConfig() const override {
+
+    static std::unique_ptr<BaseConfig>
+    StaticCreateConfig() {
```

Comment:
Actually, using `CreateConfig` as the static method and `CreateConfigForInstance` as the class method sounds better to me. Just a suggestion, no strong opinion on this

Reply:
technically, it is not just `CreateConfig`, it is `CreateConfigTemplate`. There is some design pattern on that.

Reply:
Is it just a naming difference? The plan is to standardize the naming of this type of static registration function as 'Static' to distinguish them.